### PR TITLE
JFG-959: add Next-config-to-execute header on PUT request to /envs

### PR DESCRIPTION
**Pull request by anton-antonenko**

### CODE REVIEW

File: jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java [hunk 1]
```diff
@@ -50,9 +55,12 @@
 
     private TestEnvironmentService testEnvService;
 
+    private JobExecutionService jobExecutionService;
```
Comment (a-badaev) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java]: can make them final

File: jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java [hunk 2]
```diff
@@ -97,9 +105,27 @@ public TestEnvironmentRestController(TestEnvironmentService testEnvironmentServi
...
+                setNextConfigToExecuteHeader(response, testSuiteNameToExecute);
+            }
+        }
         return ResponseEntity.accepted().build();
     }
 
+    private String getTestSuiteNameToExecute(TestEnvironmentEntity testEnv) {
+        List<String> testSuitesNames = testEnv.getTestSuites().stream().map(TestSuiteEntity::getTestSuiteId).collect(toList());
+
+        return jobExecutionService.readAll().stream()
```
Comment (a-badaev) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java]: can return Optional from here

File: jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java [hunk 3]
```diff
@@ -97,9 +105,27 @@ public TestEnvironmentRestController(TestEnvironmentService testEnvironmentServi
...
+                .map(JobExecutionEntity::getJob)
+                .filter(job -> job.getEnvId().equals(testEnv.getEnvironmentId()))
+                .map(JobEntity::getTestSuiteId)
+                .filter(testSuitesNames::contains)
+                .findFirst()
+                .orElse(null);
```
Comment (dlatnikov) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java]: are you reading all jobExecEntities or only PENDIG?

Comment (anton-antonenko) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java]: All. Should I read only pending?

File: jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java (hunk 2 above)
Comment (anton-antonenko) [inline jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java]: Ok

**Pull request merged**